### PR TITLE
CrUX API announcement

### PR DESCRIPTION
**Pull request by rviscomi**
Changes proposed in this pull request:

- new blog post and guide about using the CrUX API

cc @exterkamp

### CODE REVIEW

**Comment by rviscomi** [comment]
@robdodson two very minor platform-specific style nits observed in this post:

In the [Querying an origin](https://deploy-preview-3321--web-dev-staging.netlify.app/chrome-ux-report-api/#querying-origin-data) section I needed a way to link directly to the `CrUXApiUtil` code block. I wrapped the name of the backticked object name in the preceding paragraph in a `<a name="crux-api-util">` tag. The stylesheet understandably applied blue coloring to the object name, even though it's not clickable and is only meant to act as an anchor for deep linking to that part of the page. I also tried adding an ID attr to the code tag `<code id="crux-api-util>` but this inherits some heading-like margins.

In the [What's next?](https://deploy-preview-3321--web-dev-staging.netlify.app/chrome-ux-report-api/#what's-next) section, I have a two-deep nested unordered list. The nested list items are indented but otherwise styled identically to the outer list items. I would have expected some other kind of delineation, like `list-item` circle vs disc. It also feels like there's a lot of extra whitespace.

**Comment by kaycebasques** [comment]
@rviscomi please kick off the new content process via go/webdev-content-request next time before submitting a PR

**Comment by rviscomi** [comment]
😱 apologies, I wasn't aware of that process! (or more likely, totally forgot about it). Will definitely file that in advance next time.

The good news is this post is reasonably flexible on publication timing by about a week as needed.

Sorry again!

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 1]
```diff
@@ -0,0 +1,388 @@
...
+The `curl` command is made up of three parts:
+
+1. The URL endpoint of the API, including the caller's private API key.
+2. The `Content-Type: application/json` header, indicating that the request body contains JSON.
+3. The JSON-encoded [request body](https://developers.google.com/web/tools/chrome-user-experience-report/api/reference/rest/v1/records/queryRecord#request-body), specifying the `https://web.dev` origin.
+
+To do the same thing in JavaScript, use the <a name="crux-api-util">`CrUXApiUtil`</a> utility, which makes the API call and returns the decoded response.
+
+```js/2
+const CrUXApiUtil = {};
```
Comment (paulirish) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:51]: hmmmm now this isn't the same as https://developers.google.com/web/tools/chrome-user-experience-report/api/guides/getting-started#javascript

i like some of these changes.. maybe we can sort it out in the forthcoming PR to the crux repo? and then sync up both these other uses?

Comment (rviscomi) [reply src/site/content/en/fast/chrome-ux-report-api/index.md:51]: Yes, good call. Pinged you offline to share my ideas for deduplicating these docs.

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+  - rviscomi
+  - exterkamp
+hero: hero.png
+description: |
+  Learn how to use the Chrome UX Report API to get easy, RESTful access to 
+  real-user experience data across millions of websites.
+date: 2020-06-23
+tags:
+  - performance
+  - blog
```
Comment (kaycebasques) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:14]: Add crux tag: https://github.com/GoogleChrome/web.dev/blob/master/src/site/_data/postTags.js#L40

**Comment by kaycebasques** [comment]
@rviscomi please merge this yourself before EOD Monday 6/29 (PST)

File: src/site/_data/paths/fast.js [hunk 3]
```diff
@@ -106,6 +106,7 @@ module.exports = {
         'chrome-ux-report-data-studio-dashboard',
         'chrome-ux-report-pagespeed-insights',
         'chrome-ux-report-bigquery',
+        'chrome-ux-report-api',
```
Comment (kaycebasques) [inline src/site/_data/paths/fast.js:109]: @robdodson or @samthor

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+
+This API has been built from the ground up with the goals of providing developers with simple, fast, and comprehensive access to CrUX data. The CrUX API only reports _field-based_ user experience data, unlike the existing [PageSpeed Insights API](https://developers.google.com/speed/docs/insights/v5/get-started), which also runs _lab-based_ Lighthouse performance audits. The CrUX API is streamlined and can quickly serve user experience data, making it ideally suited for real-time auditing applications.
```
Comment (mihajlija) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:25]:
```suggestion
This API has been built with the goal of providing developers with simple, fast, and comprehensive access to CrUX data. The CrUX API only reports [_field_](/how-to-measure-speed/#lab-data-vs-field-data) user experience data, unlike the existing [PageSpeed Insights API](https://developers.google.com/speed/docs/insights/v5/get-started), which also reports _lab_ data from the Lighthouse performance audits. The CrUX API is streamlined and can quickly serve user experience data, making it ideally suited for real-time auditing applications.
```

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+API_KEY="[YOUR_API_KEY]"
+curl "https://chromeuxreport.googleapis.com/v1/records:queryRecord?key=$API_KEY" \
+  --header 'Content-Type: application/json' \
+  --data '{"origin": "https://web.dev"}'
+```
+
+Run this query interactively in the [CrUX API explorer](https://developers.google.com/web/tools/chrome-user-experience-report/api/reference/rest/v1/records/queryRecord?apix=true&apix_params=%7B%22resource%22%3A%7B%22origin%22%3A%22https%3A%2F%2Fwww.google.com%22%7D%7D).
+
+{% Aside %}
+Note that all API requests must provide a value for the `key` parameter, the placeholder for which is left as `YOUR_API_KEY`. Get your own private CrUX API key at the click of a button in the official [CrUX API documentation](https://developers.google.com/web/tools/chrome-user-experience-report/api/guides/getting-started#APIKey). For convenience, the interactive [CrUX API explorer](https://developers.google.com/web/tools/chrome-user-experience-report/api/reference/rest/v1/records/queryRecord?apix=true) does not require an API key.
```
Comment (mihajlija) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:45]:
```suggestion
All API requests must provide a value for the `key` parameter—`YOUR_API_KEY` in the example above is left as a placeholder for that value. Get your own private CrUX API key at the click of a button in the official [CrUX API documentation](https://developers.google.com/web/tools/chrome-user-experience-report/api/guides/getting-started#APIKey). For convenience, the interactive [CrUX API explorer](https://developers.google.com/web/tools/chrome-user-experience-report/api/reference/rest/v1/records/queryRecord?apix=true) does not require an API key.
```

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+{
+  "error": {
+    "code": 404,
+    "message": "chrome ux report data not found",
+    "status": "NOT_FOUND"
+  }
+}
+```
+
+To debug this error, first check that the requested origin is publicly navigable. You can test this by entering the origin into your browser's URL bar and comparing it against the final URL after any redirects. Common problems may be unnecessarily adding or omitting the subdomain or using the wrong HTTP protocol.
```
Comment (mihajlija) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:141]:
```suggestion
To debug this error, first check that the requested origin is publicly navigable. You can test this by entering the origin into your browser's URL bar and comparing it against the final URL after any redirects. Common problems include unnecessarily adding or omitting the subdomain and using the wrong HTTP protocol.
```

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+{% CompareCaption %}
+This origin is publicly navigable.
+{% endCompareCaption %}
+{% endCompare %}
+
+If the requested origin _is_ the navigable version, this error may also occur if the origin has an insufficient number of samples. All origins and URLs included in the dataset must have a sufficient number of samples to anonymize individual users. Additionally, origins and URLs must be [publicly crawlable](https://developers.google.com/search/reference/robots_txt). Refer to the [CrUX methodology](https://developers.google.com/web/tools/chrome-user-experience-report#methodology) to learn more about how websites are included in the dataset.
+
+## Querying URL data
+
+We've seen how to query the CrUX API for the overall user experience on an origin. Now let's see how to restrict the results to a particular page. To do that, we use the `url` request parameter.
```
Comment (mihajlija) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:165]:
```suggestion
You've seen how to query the CrUX API for the overall user experience on an origin. To restrict the results to a particular page, use the `url` request parameter.
```

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 8]
```diff
@@ -0,0 +1,394 @@
...
+
+The results show that this page passes the Core Web Vitals assessments for all three metrics.
+
+```text
+The 75th percentile (1973) of largest_contentful_paint passes the Core Web Vitals "good" threshold (2500).
+The 75th percentile (20) of first_input_delay passes the Core Web Vitals "good" threshold (100).
+The 75th percentile (0.05) of cumulative_layout_shift passes the Core Web Vitals "good" threshold (0.10).
+```
+
+Combined with an automated way to monitor API results, data from CrUX can be used to ensure that real-user experiences **get fast** and **stay fast**. For more information about Core Web Vitals and how to measure them, check out [Web Vitals](/vitals) and [Tools to measure Core Web Vitals](https://web.dev/vitals-tools).
```
Comment (mihajlija) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:372]:
```suggestion
Combined with an automated way to monitor API results, data from CrUX can be used to ensure that real-user experiences **get fast** and **stay fast**. For more information about Core Web Vitals and how to measure them, check out [Web Vitals](/vitals) and [Tools to measure Core Web Vitals](/vitals-tools).
```

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 9]
```diff
@@ -0,0 +1,394 @@
...
+
+## What's next?
+
+The features included in this inaugural version of the CrUX API only scratch the surface of the kinds of insights that are possible with CrUX. Users of the [CrUX dataset on BigQuery](https://web.dev/chrome-ux-report-bigquery/) may be familiar with some of the more advanced features including:
```
Comment (mihajlija) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:376]:
```suggestion
The features included in this inaugural version of the CrUX API only scratch the surface of the kinds of insights that are possible with CrUX. Users of the [CrUX dataset on BigQuery](/chrome-ux-report-bigquery/) may be familiar with some of the more advanced features including:
```

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 10]
```diff
@@ -0,0 +1,394 @@
...
+
+This API has been built from the ground up with the goals of providing developers with simple, fast, and comprehensive access to CrUX data. The CrUX API only reports _field-based_ user experience data, unlike the existing [PageSpeed Insights API](https://developers.google.com/speed/docs/insights/v5/get-started), which also runs _lab-based_ Lighthouse performance audits. The CrUX API is streamlined and can quickly serve user experience data, making it ideally suited for real-time auditing applications.
+
+To ensure that developers have access to all of the metrics that matter most—[Largest Contentful Paint](https://web.dev/lcp/) (LCP), [First Input Delay](https://web.dev/fid/) (FID), and [Cumulative Layout Shift](https://web.dev/cls/) (CLS)—the CrUX API audits and monitors these [Core Web Vitals](https://web.dev/vitals/#core-web-vitals) at both the origin and URL level.
```
Comment (mihajlija) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:27]:
```suggestion
To ensure that developers have access to all of the metrics that matter the most—the [Core Web Vitals](/vitals/#core-web-vitals)—the CrUX API audits and monitors [Largest Contentful Paint](/lcp/) (LCP), [First Input Delay](/fid/) (FID), and [Cumulative Layout Shift](/cls/) (CLS) at both the origin and URL level.
```

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 11]
```diff
@@ -0,0 +1,394 @@
...
+          "p75": 2216
+        }
+      },
+      // ...
+    }
+  }
+}
+```
+
+The `start` and `end` properties of the `histogram` object represent the range of values users experience for the given metric. The `density` property represents the proportion of user experiences within that range. In this example, 79% of LCP user experiences across all web.dev pages are under 2,500 milliseconds, which is the "[good](https://web.dev/lcp/#what-is-lcp)" LCP threshold. The `percentiles.p75` value means that 75% of user experiences in this distribution are less than 2,216 milliseconds. Learn more about the response structure in the [response body](https://developers.google.com/web/tools/chrome-user-experience-report/api/reference/rest/v1/records/queryRecord#response-body) documentation.
```
Comment (mihajlija) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:125]:
```suggestion
The `start` and `end` properties of the `histogram` object represent the range of values users experience for the given metric. The `density` property represents the proportion of user experiences within that range. In this example, 79% of LCP user experiences across all web.dev pages are under 2,500 milliseconds, which is the "[good](/lcp/#what-is-lcp)" LCP threshold. The `percentiles.p75` value means that 75% of user experiences in this distribution are less than 2,216 milliseconds. Learn more about the response structure in the [response body](https://developers.google.com/web/tools/chrome-user-experience-report/api/reference/rest/v1/records/queryRecord#response-body) documentation.
```

File: src/site/content/en/fast/chrome-ux-report-api/index.md [hunk 12]
```diff
@@ -0,0 +1,394 @@
...
+
+{% Aside 'caution' %}
+The more fine-grained the request is, for example a specific combination of URL and form factor, the fewer user experiences it will include. This may lead to more frequent "not found" errors, especially when querying less popular URLs or the less popular tablet device type.
+{% endAside %}
+
+Recall from the previous section that 85% of user experiences on this page had "good" LCP. Compare that to phone-specific experiences, of which only 78% are considered "good". The 75th percentile is also slower among phone experiences, up from 1,947 milliseconds to 2,366 milliseconds. Segmenting by form factor has the potential to highlight more extreme disparities in user experiences.
+
+## Assessing Core Web Vitals performance
+
+The [Core Web Vitals](https://web.dev/vitals/#core-web-vitals) program defines targets that help determine whether a user experience or a distribution of experiences can be considered "good". In the following example, we use the CrUX API and the [`CrUXApiUtil.query`](#crux-api-util) function to assess whether a web page's distribution of Core Web Vitals metrics (LCP, FID, CLS) are "good".
```
Comment (mihajlija) [inline src/site/content/en/fast/chrome-ux-report-api/index.md:324]:
```suggestion
The [Core Web Vitals](/vitals/#core-web-vitals) program defines targets that help determine whether a user experience or a distribution of experiences can be considered "good". In the following example, we use the CrUX API and the [`CrUXApiUtil.query`](#crux-api-util) function to assess whether a web page's distribution of Core Web Vitals metrics (LCP, FID, CLS) are "good".
```

**Comment by mihajlija** [comment]
@rviscomi Heya, Kayce asked me to review this. I've suggested a few edits that might help brevity and clarity and I've replaced all web.dev links with relative URLs. Other than that, LGTM :)

File: src/site/content/en/fast/chrome-ux-report-api/index.md (hunk 5 above)
Comment (rviscomi) [reply src/site/content/en/fast/chrome-ux-report-api/index.md]: Accepted suggestion with a change to use the shortened API key URL.

File: src/site/_data/paths/fast.js (hunk 3 above)
Comment (rviscomi) [reply src/site/_data/paths/fast.js:109]: Removing this change to unblock the PR.

**Pull request merged**